plugins/pulp_rpm/plugins/importers/iso/sync.py

```diff
@@ -229,7 +234,9 @@ def perform_sync(self):
                     iso.save()
                 except NotUniqueError:
                     iso = iso.__class__.objects.filter(**iso.unit_key).first()
-                self.add_catalog_entries([iso])
+                else:
+                    # Only if iso is from pulp manifest
```

Comment:
Well, when we have ``NotUniqueError`` it means the iso is already in Pulp and therefore does not have the special parameter url from pulp manifest. and we just associate it to repo. Basically it is just optimisation to not call add_catalog_entries which would always do nothing.